ihmc-robotics-toolkit/src/main/java/us/ihmc/robotics/screwTheory/FloatingBaseRigidBodyDynamicsCalculator.java

```diff
@@ -4,133 +4,281 @@
-         DMatrixRMaj qddot, DMatrixRMaj tau, DMatrixRMaj rho)
+   /**
+    * The rigid body dynamics are
+    * <p>
+    * [0; &tau;<sup>T</sup>]<sup>T</sup> = H(q) qDdot + C(q, qDot) + J<sub>c</sub><sup>T</sup> &rho;
+    * </p>
+    * <p>
+    * The actuated portion of the dynamics are then:
+    * </p>
+    * <p>
+    * &tau; = H<sub>body</sub>(q) qDdot + C<sub>body</sub>(q,qDot) + J<sub>body</sub><sup>T</sup> &rho;
+    * </p>
+    * <p>
+    * This method performs the above calculation and checks whether the evaluation equals zero.
+    * </p>
+    *
+    * @param bodyMassMatrix                 bottom N rows of the H(q) matrix above, which is H<sub>body</sub>(q). Not modified.
+    * @param bodyCoriolisMatrix             bottom N rows of the C(q, qDot) matrix above, which is C<sub>body</sub>(q, qDot). Not modified.
+    * @param bodyContactForceJacobianMatrix right N columns of the J matrix above, which is J<sub>body</sub>. Not modified.
+    * @param jointAccelerations             joint accelerations. &rho; in the above equations. Not modified.
+    * @param jointTorques                   joint torques. &tau; in the above equations. Not modified.
+    * @param contactForces                  contact forces. qDdot in the above equations. Not modified.
+    * @return where the defined dynamics hold.
+    */
+   public boolean areActuatedDynamicsSatisfied(DMatrixRMaj bodyMassMatrix,
+                                               DMatrixRMaj bodyCoriolisMatrix,
+                                               DMatrixRMaj bodyContactForceJacobianMatrix,
+                                               DMatrixRMaj jointAccelerations,
+                                               DMatrixRMaj jointTorques,
+                                               DMatrixRMaj contactForces)
    {
-      computeJacobianTranspose(bodyContactForceJacobianMatrix, matrixTranspose);
-
-      CommonOps_DDRM.multAdd(bodyMassMatrix, qddot, bodyCoriolisMatrix);
-      CommonOps_DDRM.multAdd(-1.0, matrixTranspose, rho, bodyCoriolisMatrix);
-      CommonOps_DDRM.subtractEquals(bodyCoriolisMatrix, tau);
-
-      return equalsZero(bodyCoriolisMatrix, tolerance);
-   }
+      tempObjective.set(bodyCoriolisMatrix);
+      CommonOps_DDRM.multAdd(bodyMassMatrix, jointAccelerations, tempObjective);
+      CommonOps_DDRM.multAddTransA(bodyContactForceJacobianMatrix, contactForces, tempObjective);
+      CommonOps_DDRM.subtractEquals(tempObjective, jointTorques);
 
-   private void computeJacobianTranspose(DMatrixRMaj jacobian, DMatrixRMaj jacobianTransposeToPack)
-   {
-      jacobianTransposeToPack.reshape(jacobian.getNumCols(), jacobian.getNumRows());
-      jacobianTransposeToPack.zero();
-      CommonOps_DDRM.transpose(jacobian, jacobianTransposeToPack);
+      return equalsZero(tempObjective, tolerance);
    }
 
    private static boolean equalsZero(DMatrixRMaj matrix, double tolerance)
```

Comment:
Can you not use `MatrixFeatures_DDRM.isZeros`?

Reply:
I didn't know that was a thing.